### ISSUE TITLE
Regular expression should allow escape chars

### ISSUE THREAD

**Issue by zehavibarak** [eslint/eslint#13446]
The code

```ts
  readonly pattern = /^((\w+(\,\w+)*)|(\w+\-\w+)|([\w\.\%\*\#]+))([;]((\w+(\,\w+)*)|(\w+\-\w+)|([\w\.\%\*\#]+)))*$/;
```

yields the warning 

> Unnecessary escape character

while it should allow it.
